File: ts/src/LngLatBBox.ts
import Constant from './Constant';

export default class LngLatBBox {
  private west: number;
  private south: number;
  private east: number;
  private north: number;

  constructor(west: number, south: number, east: number, north: number) {
    this.west = west;
    this.south = south;
    this.east = east;
    this.north = north;
  }

  public getEast() {
    return this.east;
  }

  public getNorth() {
    return this.north;
  }

  public getSouth() {
    return this.south;
  }

  public getWest() {
    return this.west;
  }

  public setEast(east: number) {
    this.east = east;
    return this;
  }

  public setNorth(north: number) {
    this.north = north;
    return this;
  }

  public setSouth(south: number) {
    this.south = south;
    return this;
  }

  public setWest(west: number) {
    this.west = west;
    return this;
  }

  public equals(o: LngLatBBox) {
    if (o === this) {
      return true;
    }

    return (
      Constant.closeTo(o.getEast(), this.getEast()) &&
      Constant.closeTo(o.getNorth(), this.getNorth()) &&
      Constant.closeTo(o.getWest(), this.getWest()) &&
      Constant.closeTo(o.getSouth(), this.getSouth())
    );
  }
}

File: java/src/main/java/com/sakitam/mercantile/LngLatBBox.java
package com.sakitam.mercantile;

import java.io.Serializable;

public class LngLatBBox implements Serializable {
    private static final long serialVersionUID = 1L;

    private double west;

    private double south;
    private double east;
    private double north;

    public LngLatBBox() {
        super();
    }

    public LngLatBBox(double west, double south, double east, double north) {
        super();
        this.west = west;
        this.south = south;
        this.east = east;
        this.north = north;
    }

    public double getEast() {
        return east;
    }

    public double getNorth() {
        return north;
    }

    public double getSouth() {
        return south;
    }

    public double getWest() {
        return west;
    }

    public void setEast(double east) {
        this.east = east;
    }

    public void setNorth(double north) {
        this.north = north;
    }

    public void setSouth(double south) {
        this.south = south;
    }

    public void setWest(double west) {
        this.west = west;
    }

    public boolean equals(LngLatBBox o) {
        if (o == this) {
            return true;
        }

        return Constant.closeTo(o.getEast(), this.getEast())
                && Constant.closeTo(o.getNorth(), this.getNorth())
                && Constant.closeTo(o.getWest(), this.getWest())
                && Constant.closeTo(o.getSouth(), this.getSouth());
    }
}
